### PR TITLE
Mask account key for azureBlobStorage

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Updated to mask account key in logs in azureBlobStorage

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit b48a20735fe8ada2b02684a283331627b264a2e6 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65273/b48a20735fe8ada2b02684a283331627b264a2e6/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Parsers/FunctionSecretArgumentsFinderAST.h [hunk 1]
```diff
@@ -169,6 +179,40 @@ class FunctionSecretArgumentsFinderAST
...
+        /// azureBlobStorage(connection_string|storage_account_url, container_name, blobpath, account_name, account_key, format, compression, structure)
+        /// azureBlobStorageCluster(cluster, connection_string|storage_account_url, container_name, blobpath, [account_name, account_key, format, compression, structure])
+        size_t count = excludeS3OrURLNestedMaps();
+        if ((url_arg_idx + 4 <= count) && (count <= url_arg_idx + 7))
+        {
+            String second_arg;
+            if (tryGetStringFromArgument(url_arg_idx + 3, &second_arg))
+            {
+                if (boost::iequals(second_arg, "NOSIGN"))
+                    return; /// The argument after 'url' is "NOSIGN".
```
Comment (kssenii) [inline src/Parsers/FunctionSecretArgumentsFinderAST.h:204]: AFAIK we do not support NOSIGN keyword for azure, do we need this?

File: src/Parsers/FunctionSecretArgumentsFinderAST.h [hunk 2]
```diff
@@ -169,6 +179,40 @@ class FunctionSecretArgumentsFinderAST
...
+        {
+            /// azureBlobStorage(named_collection, ..., account_key = 'account_key', ...)
+            findSecretNamedArgument("account_key", 1);
+            return;
+        }
+
+        /// We should check other arguments first because we don't need to do any replacement in case storage_account_url is not used
+        /// azureBlobStorage(connection_string|storage_account_url, container_name, blobpath, account_name, account_key, format, compression, structure)
+        /// azureBlobStorageCluster(cluster, connection_string|storage_account_url, container_name, blobpath, [account_name, account_key, format, compression, structure])
+        size_t count = excludeS3OrURLNestedMaps();
```
Comment (kssenii) [inline src/Parsers/FunctionSecretArgumentsFinderAST.h:197]: Also AFAIK we do not support `headers` and `extra_credentials` for azure as well

File: src/Parsers/FunctionSecretArgumentsFinderAST.h [hunk 3]
```diff
@@ -169,6 +179,40 @@ class FunctionSecretArgumentsFinderAST
             markSecretArgument(url_arg_idx + 2);
     }
 
+    void findAzureBlobStorageFunctionSecretArguments(bool is_cluster_function)
+    {
+        /// azureBlobStorage('cluster_name', 'conn_string/storage_account_url', ...) has 'conn_string/storage_account_url' as its second argument.
+        size_t url_arg_idx = is_cluster_function ? 1 : 0;
+
+        if (!is_cluster_function && isNamedCollectionName(0))
```
[the remark this answers is not in the document]
Comment (SmitaRKulkarni) [reply src/Parsers/FunctionSecretArgumentsFinderAST.h:187]: good point, it was not working with `is_cluster_function = true`, fixed it and added tests

File: src/Parsers/FunctionSecretArgumentsFinderAST.h (hunk 1 above)
Comment (SmitaRKulkarni) [reply src/Parsers/FunctionSecretArgumentsFinderAST.h]: not needed, removed it

File: src/Parsers/FunctionSecretArgumentsFinderAST.h (hunk 2 above)
Comment (SmitaRKulkarni) [reply src/Parsers/FunctionSecretArgumentsFinderAST.h]: removed it

**Review by kssenii — APPROVED** [review]

**Comment by SmitaRKulkarni** [comment]
Unrelated failures : 
Integration tests (tsan) [2/6] : test_storage_s3_queue/test.py::test_max_set_age
Integration tests (tsan) [4/6] : test_materialized_mysql_database/test.py::test_materialized_database_mysql_date_type_to_date32
Stress test (tsan) : S3_ERROR No such key thrown

**Pull request merged**